File: RotateImage.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        self.transpose(matrix)
        for row in matrix:
            row.reverse()
        return matrix

    def transpose(self, matrix: List[List[int]]) -> None:
        for i in range(len(matrix[0])):
            for j in range(i + 1, len(matrix[0])):
                matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
        for i in matrix:
            for j in i:
                print(j, end=" ")
            print()
        return matrix

File: RotateImage.cpp
#include <algorithm>
#include <vector>

class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        // Tranpose matrix
        for (int i = 0; i < matrix[0].size(); i++) {
            for (int j = i + 1; j < matrix[0].size(); j++) {
                int temp = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = temp;
            } 
        }

        // Reverse each row in the matrix
        for (int i = 0; i < matrix[0].size(); i++) {
            std::reverse(matrix[i].begin(), matrix[i].end());
        }
    }
};
